src/fheroes2/dialog/dialog_resolution.cpp

```diff
@@ -246,21 +292,36 @@ namespace Dialog
                 fheroes2::showMessage( header, body, 0 );
             }
 
-            if ( resList.isSelected() ) {
-                selectedResolution = resList.GetCurrent();
+            if ( !listBox.IsNeedRedraw() && !needRedraw ) {
```

Comment:
The same comment here :)
```suggestion
            if ( !listBox.IsNeedRedraw() ) {
```